airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -195,10 +199,12 @@ def __init__(self):
         self._validate()
 
     def _validate(self):
-        if not self.dags_volume_claim and (not self.git_repo or not self.git_branch):
+        if not self.dags_volume_claim and not self.dags_in_docker \
+                and (not self.git_repo or not self.git_branch):
```

Comment:
this would collide with the changes in https://github.com/apache/incubator-airflow/pull/3770/files#diff-b1d8d65aeaa7d031dfe5b197d6c5aa69L200